### PR TITLE
Chat: add search by company name/representative name

### PR DESCRIPTION
**Pull request by lilsweetcaligula**
#4 

This is a rather simplistic implementation of the search. Please let me know if you want the search to be more advanced.

P.S. I tested the styling on Firefox. Could you please check the style on Chrome for me? Thing is if I run Chrome along with Firefox on my ancient machine, I fear it's gonna take the CPU back to 264 B.C.

### CODE REVIEW

File: src/candidates/profile/chat/shared/navigation/ChatNavigation.vue [hunk 1]
```diff
@@ -69,7 +68,28 @@ export default {
...
       selectContact: 'chat/selectContact'
-    })
+    }),
+    filterItemsMatchingSearch (list) {
+      if (!this.searchText) {
+        return list;
+      }
+
+      const includesIgnoreCase = (str, suffix) =>
+        str.toLowerCase().includes(suffix.toLowerCase());
```
Comment (lilsweetcaligula) [inline src/candidates/profile/chat/shared/navigation/ChatNavigation.vue]: I was initially considering extracting this function as a utility into the core module, but ultimately found it more readable to inline it here.

File: src/candidates/profile/chat/shared/navigation/ChatNavigation.vue [hunk 2]
```diff
@@ -69,7 +68,28 @@ export default {
...
+
+      return list.filter((item) => {
+        const company = item.company.name;
+        const rep = item.latestRepresentative;
+
+        const matchesCompany = includesIgnoreCase(company, this.searchText);
+        const matchesRepresentative = includesIgnoreCase(rep, this.searchText);
```
Comment (lilsweetcaligula) [inline src/candidates/profile/chat/shared/navigation/ChatNavigation.vue]: My hands were all kinds of itchy to replace `includesIgnoreCase` with something like
```
const matchesSearchText = (str) =>
  str.toLowerCase().includes(this.searchText.toLowerCase());
```

Then this code would look as follows:
```
const matchesCompany = matchesSearchText(company);
const matchesRepresentative = matchesSearchText(rep);
```

However, I ultimately decided that the extra-fanciness would come at the cost of the straightforwardness of the code.

**Comment by lilsweetcaligula** [comment]
I just noticed that the Messages page is missing a header (along with the header menu and a greeting). Is that by design?

**Comment by tetiana-kolmychek-devv** [comment]
@lilsweetcaligula yes, missing header in chat was made by design to free more space for messages.
Do you think it would be beneficial to have it there?

**Comment by lilsweetcaligula** [comment]
@tetiana-kolmychek-devv I believe so, because it will make design more consistent across pages.

**Comment by tetiana-kolmychek-devv** [comment]
@lilsweetcaligula I checked your PR, everything seems to be working fine on both desktop and mobile view.

Just one question: when user switches to another tab, search is still active and filters results on the new active tab. Is it by design?

I don't know which is more user-friendly: leave it as it is or empty search on tab switch, so tell me your opinion. :)

**Comment by lilsweetcaligula** [comment]
@tetiana-kolmychek-devv Resetting search on tab switch could be more predictable. I can fix that.

**Comment by tetiana-kolmychek-devv** [comment]
@lilsweetcaligula Would be great! Will wait for fix to review.

**Comment by lilsweetcaligula** [comment]
@tetiana-kolmychek-devv I looked into the issue. I think I misunderstood you. Sorry about that.

I thought "tabs" were referring to the pages on the site.

If "tabs" are referring to the browser tabs, then I think it's better to keep it as is. As an example, if one starts to type a query on the Google page, then switches to another tab, then goes back, - the query in the input field will not be reset.

**Pull request merged**